src/utils/timestampToEpoch.ts

```diff
@@ -0,0 +1,49 @@
+import { BigNumberish } from "ethers";
+import { AgeConfig, ages, EpochConfig } from "../ages";
+import { Optional } from "../helpers/types";
+
+export const timestampToEpoch = (timestamp: BigNumberish) => {
+  const epoch = allEpochs.find((epoch) => epoch.initialTimestamp.lte(timestamp) && epoch.finalTimestamp.gt(timestamp));
+  if (!epoch) return;
+  const age = ages[epoch.ageId];
+  return {
+    age: age as AgeConfig<EpochConfig>,
+    epoch: age.epochs[epoch.epochId] as EpochConfig,
+  };
+};
+const allEpochs = ages
+  .map((age, ageId) => age.epochs.map((epoch, epochId) => ({ ...epoch, age: age.ageName, ageId, epochId })))
+  .flat();
+
+export const getEpochsBetweenTimestamps = (t1: BigNumberish, t2: BigNumberish) => {
+  const epoch1 = timestampToEpoch(t1);
+  if (!epoch1) return;
+  const epoch2 = timestampToEpoch(t2);
+  const epochs = [epoch1];
+  if (!epoch2) return epochs;
```

Comment:
```suggestion
export const getEpochsBetweenTimestamps = (tFrom: BigNumberish, tTo: BigNumberish) => {
  const epochFrom = timestampToEpoch(tFrom);
  if (!epochFrom) return;
  const epochTo = timestampToEpoch(tTo);
  const epochs = [epochFrom];
  if (!epochTo) return epochs;
```

Otherwise it can be confusing with the epochs count